components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -478,6 +489,35 @@ && isStepHasMultiOption(context)
         }
     }
 
+    protected boolean isApplicationEnabled(HttpServletRequest request, AuthenticationContext context)
+            throws FrameworkException {
+
+        String type = request.getParameter(TYPE);
+        String clientId = request.getParameter(FrameworkConstants.RequestParams.ISSUER);
+        if (StringUtils.isBlank(type)) {
+            type = context.getRequestType();
+        }
+        if (StringUtils.isBlank(clientId)) {
+            clientId = request.getParameter(CLIENT_ID);
+        }
+        if (StringUtils.isBlank(clientId)) {
+            clientId = context.getRelyingParty();
+        }
+
+        ServiceProvider serviceProvider = getServiceProvider(type, clientId, getTenantDomain(request));
+        if (serviceProvider == null) {
+            return false;
```

Comment:
instead of returning false lets throw an exception the same way how we would throw an exception in the auth flow if the app is not available in other places. Since we are unable to determine the state of the app flaging as disabled could make it hard when we need to debug.